api/v1alpha1/thanosquery_types.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ThanosQuerySpec defines the desired state of ThanosQuery
+type ThanosQuerySpec struct {
+	CommonThanosFields `json:",inline"`
+
+	// Querier replica labels to configure.
+	QuerierReplicaLabels []string `json:"replicaLabels,omitempty"`
+
+	// Override default StoreAPI selector labels, to select specific StoreAPIs to query.
+	CustomStoreLabelSelector *metav1.LabelSelector `json:"storeLabelSelector,omitempty"`
```

Comment:
might be worth adding the kubebuilder optional marker on these just for clarity.

also might be worth either commenting on or specifying the default behaviour